src/functions-and-arrays.js

```diff
@@ -1,41 +1,112 @@
 // Iteration #1: Find the maximum
-function maxOfTwoNumbers() {}
 
+function maxOfTwoNumbers(num1, num2) {
+  let bigNumber;
+
+  if (num1 > num2) {
+      bigNumber = num1;
+  } else {
+      bigNumber = num2;
+  }
+
+  return bigNumber;
+}
 
 
 // Iteration #2: Find longest word
 const words = ['mystery', 'brother', 'aviator', 'crocodile', 'pearl', 'orchard', 'crackpot'];
 
-function findLongestWord() {}
+function findLongestWord(words) {
+    if (words.length === 0) {
+      return null;
+    }
+
+    if (words.length === 1) {
+      return words[0]
+    }
+
+    let longestWord = '';
+
+        for (let i = 0; i < words.length; i++) {
+            const findWord = words[i];
+
+            if (findWord.length > longestWord.length || longestWord === '') {
+                longestWord = findWord;
+            }
+
+        }
+
+    return longestWord;
+}
 
 
 
 // Iteration #3: Calculate the sum
 const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
 
-function sumNumbers() {}
+function sumNumbers(numbers) {
+    if (numbers.length === 0) {
+        return 0;
+      }
+   
+    let sum = 0;
+
+    for (let i = 0; i < numbers.length; i++) {
+        sum += numbers[i];
+        
+    }
+    return sum;
+}
 
 
 
 // Iteration #3.1 Bonus:
-function sum() {}
+//function sum() {}
 
 
 
 // Iteration #4: Calculate the average
 // Level 1: Array of numbers
 const numbersAvg = [2, 6, 9, 10, 7, 4, 1, 9];
 
-function averageNumbers() {}
+function averageNumbers(numbers) {
+  if (numbers.length === 0) {
+    return null;
+  }
+  let sum = 0;
+
+  for (let i = 0; i < numbers.length; i++) {
+    sum += numbers[i];
+  }
+
+  const average = sum/ numbers.length;
+  return average;
+}
 
 
 // Level 2: Array of strings
 const wordsArr = ['seat', 'correspond', 'linen', 'motif', 'hole', 'smell', 'smart', 'chaos', 'fuel', 'palace'];
 
-function averageWordLength() { }
+function averageWordLength(words) {
+  if (words.length === 0) {
+    return null;
+  }
+
+  let totalLenght = 0;
+
+  for (let i = 0; i < words.length; i++) {
+    totalLenght += words[i].length;
+  }
+
+  const averageLength = totalLenght / words.length;
+  return averageLength;
+
+ }
+
+
 
 // Bonus - Iteration #4.1
-function avg() {}
+//function avg() {}
```

Comment:
```suggestion
function avg() {}
```
No te hace falta comentarlo, asi uno de los test no te falla jejeje